message.go

```diff
@@ -175,3 +175,14 @@ func (messages *messages) copyInternal() []Message {
 	}
 	return copiedMessages
 }
+
+// Returns all messages and removes them at the same time
+func (messages *messages) Purge() []Message {
```

Comment:
Seems this method should be unexported (private) as we are going to provide only one entrypoint for this - `server.MessagesAndPurge()`.

Reply:
Ah, true. I missed that.